server/rtcd.go

```diff
@@ -706,3 +708,51 @@ func (h *rtcdHost) isFlagged() bool {
 	defer h.mut.RUnlock()
 	return h.flagged
 }
+
+func (m *rtcdClientManager) hasCallEnded(call *public.Call) bool {
+	m.ctx.LogDebug("RTCD host is set in call, checking...", "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+	host := m.getHost(call.Props.RTCDHost)
+	if host != nil {
+		m.ctx.LogDebug("RTCD host found", "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+
+		// Version compatibility check. We need to be talking to RTCD v1.0.0 or higher to be able to call GetSessions.
+		info, err := host.client.GetVersionInfo()
+		if err != nil {
+			m.ctx.LogDebug("failed to get version info", "err", err.Error(), "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+			return false
+		}
+
+		// Always support dev builds.
+		if info.BuildVersion == "" || info.BuildVersion == "master" || strings.HasPrefix(info.BuildVersion, "dev") {
+			m.ctx.LogDebug("skipping version compatibility check", "buildVersion", info.BuildVersion, "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+		} else if err := checkMinVersion("v1.0.0", info.BuildVersion); err != nil {
+			m.ctx.LogDebug("RTCD host version is not compatible", "err", err.Error(), "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+			return false
+		}
+
+		cfgs, code, err := host.client.GetSessions(call.ID)
+		if err != nil {
+			m.ctx.LogDebug("failed to get sessions for call", "err", err.Error(), "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+		}
+
+		if code != http.StatusOK && code != http.StatusNotFound {
+			m.ctx.LogDebug("unexpected status code from RTCD", "code", code, "callID", call.ID, "rtcdHost", call.Props.RTCDHost)
+			// The request above could fail for various reasons, in which case we can't assume the call has ended.
```

Comment:
ah, this confirms the default behavior. Maybe add this comment to the above as well? (would just save some cognitive overhead)